### ISSUE TITLE
Add git hook to force usage of Conventional Commits

### ISSUE THREAD

**Issue by teo8192** [Frixxie/items#37]
Put binary/script in .git/hooks/commit-msg
Return status 0 if commit ok, non-zero for failure.
The program receives file containing commit msg as first arg.

Investigate how to make this hook follow the repo.

**Comment by teo8192** [comment]
Investigate git template directory, as the template will manage some default files that are copied into the `.git` directory.

**Comment by Frixxie** [comment]
I disagree here, we do not really need conventional commits on EVERYTHING, only the important stuff. I think it would be fine to only use it when we want it in the changelog

**Comment by teo8192** [comment]
Another advantage of conventional commits (CC?) in addition to SemVer is that the commit history is easier to explore and more readable.
CC for semver only needs fix/feat/BREAKING CHANGE, but CC also allows other types for non-digital history inspectors such as docs/refactor/perf/whatever we want to categorize the type of the commit (see [point 4](https://www.conventionalcommits.org/en/v1.0.0/)).

If categorizing a commit is only mandatory if it is «important», one would also have to decide if the commit is important, however I think it would be easy to categorize all commits if we decide on good categories/types. Suggest to start out with categories outlined in [@commlint/config-conventional](https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional):
```

[
  'build',
  'chore',
  'ci',
  'docs',
  'feat',
  'fix',
  'perf',
  'refactor',
  'revert',
  'style',
  'test'
];
```

Thoughts?

**Comment by Frixxie** [comment]
Sure, we can use it. It would be nice tho to have the validation done on the remote instead of placing it in the `.git` folder

**Comment by Frixxie** [comment]
Another way to do it is to have it as an part of ci/cd
